src/index/ivf/ivf.cc

```diff
@@ -224,9 +225,17 @@ class IvfIndexNode : public IndexNode {
     Status
     SerializeImpl(BinarySet& binset, IVFFlatTag) const;
 
+    Status
+    TrainInternal(const DataSet& dataset, const Config& cfg);
```

Comment:
I'd move this function into `private` section, because it is Internal, right? :)